### ISSUE TITLE
JsSHA is not a constructor

### ISSUE THREAD

**Issue by Maria-Elena-98** [bellstrand/totp-generator#23]
When I try to use totp, I get an error inside the index file of the totp-generator module in the getToken function

the mistake is JsSHA is not a constructor
```
'use strict';

let JsSHA = require('jssha');

module.exports = function getToken(key, options) {
	options = options || {};	
	let epoch, time, shaObj, hmac, offset, otp;
	options.period = options.period || 30;
	options.algorithm = options.algorithm || 'SHA-1';
	options.digits = options.digits || 6;
	key = base32tohex(key);
	epoch = Math.round(Date.now() / 1000.0);
	time = leftpad(dec2hex(Math.floor(epoch / options.period)), 16, '0');


	shaObj = new JsSHA(options.algorithm, 'HEX');  //          IN THIS LINE


	shaObj.setHMACKey(key, 'HEX');
	shaObj.update(time);
	hmac = shaObj.getHMAC('HEX');
	offset = hex2dec(hmac.substring(hmac.length - 1));
	otp = (hex2dec(hmac.substr(offset * 2, 8)) & hex2dec('7fffffff')) + '';
	otp = otp.substr(otp.length - options.digits, options.digits);
	return otp;
}
```

I call it this way

```
const totp = require("totp-generator");

const token = totp("JBSWY3DPEHPK3PXP");
```

I am using react, this is the package.json
```

{
  "name": "shop",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "bootstrap": "^4.6.0",
    "jssha": "^3.2.0",
    "react": "^17.0.2",
    "react-bootstrap": "^1.6.0",
    "react-dom": "^17.0.2",
    "react-redux": "^7.2.4",
    "react-router-dom": "^5.2.0",
    "react-scripts": "4.0.3",
    "totp-generator": "0.0.9",
    "web-vitals": "^1.0.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/core": "^7.14.3",
    "@babel/preset-env": "^7.14.2",
    "@babel/preset-react": "^7.13.13"
  }
}

```

**Comment by bellstrand** [comment]
@Maria-Elena-98 This repository is old and I've got no plan on developing or maintaining it since there are better alternatives out today.
I just haven't had the time to deprecate the package yet. But that will happen sooner or later.

**Comment by joegasewicz** [comment]
I just came across this bug as well.

**Comment by TheStaticTurtle** [comment]
I'm curious as what you refer as better alternative, it took quite a while until I found one that remotely meets my usage 😅.
Anyway, the fix seems pretty simple after messing arround in the console I'm pretty sure you could just do `let JsSHA = require("jssha").default` if the first one fails

**Comment by bellstrand** [comment]
@TheStaticTurtle I'd accept a PR and release a new version if you want to fix it!
I've been using https://github.com/yeojz/otplib for my last projects, but there are multiple solutions available depending on what you need!

**Comment by TheStaticTurtle** [comment]
Thanks, did a quick PR, otplib seems nice I probably missed it in the npm repo

**Comment by bellstrand** [comment]
@TheStaticTurtle Released in `v0.0.13`!

**Comment by TheStaticTurtle** [comment]
Thanks tested the release it and it works fine
